.github/workflows/asan-test.yaml

```diff
@@ -0,0 +1,39 @@
+name: ASAN Tests
+
+on: [push, pull_request]
+
+env:
+  OPEN_UCX_LINK: https://github.com/openucx/ucx
+  OPEN_UCX_BRANCH: master
+  CLANG_VER: 17
+
+jobs:
+  gtest-asan:
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
```

Comment:
It's better to move long shell scripts to a separate .sh file and call it from the workflow.

Reply:
here we launch gtests and wait for result, why it would better to wrap it to separate sh?

Reply:
There're some motivations:
- make yaml workflow concise for better readability and maintainability
- to be able to verify shell scripts with shellcheck to fix static code scanning issues